src/plugins/midi-piano/midi-piano-display.js

```diff
@@ -0,0 +1,292 @@
+import axios from 'axios';
+import * as Tone from 'tone';
+import { Button } from 'antd';
+import midiPlayerNs from 'midi-player-js';
+import urlUtils from '../../utils/url-utils.js';
+import React, { useEffect, useRef } from 'react';
+import PianoComponent from './piano-component.js';
+import { QuestionCircleOutlined } from '@ant-design/icons';
+import ClientConfig from '../../bootstrap/client-config.js';
+import { useService } from '../../components/container-context.js';
+import { sectionDisplayProps } from '../../ui/default-prop-types.js';
+import PlayIcon from '../../components/icons/media-player/play-icon.js';
+import PauseIcon from '../../components/icons/media-player/pause-icon.js';
+
+export default function MidiPianoDisplay({ content }) {
+
+  const keys = useRef([]);
+  const player = useRef(null);
+  const sampler = useRef(null);
+  const midiData = useRef(null);
+  const midiAccessObj = useRef(null);
+  const samplerHasLoaded = useRef(false);
+  const { NOTES } = midiPlayerNs.Constants;
+  const { sourceType, sourceUrl, midiTrackTitle, noteRange } = content;
+  const hasMidiFile = sourceUrl !== '';
+  const hasMidiTrackTitle = midiTrackTitle !== '';
+  const clientConfig = useService(ClientConfig);
+  const src = urlUtils.getMidiUrl({ cdnRootUrl: clientConfig.cdnRootUrl, sourceType, sourceUrl });
+
+  const getNoteNameFromMidiValue = midiValue => {
+    return NOTES[midiValue];
+  };
+
+  const getEventTypefromMidiCommand = (command, velocity) => {
+    switch (command) {
+      case 144:
+        if (velocity > 0) {
+          return 'Note on';
+        }
+        return 'Note off';
+      case 128:
+        return 'Note off';
+      case 176:
```

Comment:
These numbers look magical, it would be good to extract a constant in the `constants.js` of the plugin, something like:
```
const MIDI_COMMAND = {
    <clearName>: 144,
    <clearName>: 128,
    <clearName>: 176
}
```
This way when you read the code of `getEventTypefromMidiCommand` you can follow it nicely, like you'd read a book, and understand the logic :) 

Btw, small typo in the handler name -> getEventType**F**romMidiCommand